Source/WebCore/dom/ContainerNodeAlgorithms.cpp

```diff
@@ -110,9 +110,11 @@ static RemovedSubtreeObservability notifyNodeRemovedFromDocument(ContainerNode&
     RemovedSubtreeObservability observability = RemovedSubtreeObservability::NotObservable;
     for (RefPtr currentNode = &node; currentNode; currentNode = NodeTraversal::next(*currentNode)) {
         currentNode->removedFromAncestor(Node::RemovalType { /* disconnectedFromDocument */ true, treeScopeChange == TreeScopeChange::Changed }, oldParentOfRemovedTree);
+        if (canDelayNodeDeletion == CanDelayNodeDeletion::Yes)
+            node.document().nodesToAsyncDelete().canNodeBeAsyncDeleted(*currentNode);
```

Comment:
I don't think we should update document-global state like this.
We should make this function return an enum value in addition to RemovedSubtreeObservability.
e.g. this function could return `std::pair<RemovedSubtreeObservability, RemovedSubtreeDelayedDeletionType>`
where `RemovedSubtreeDelayedDeletionType` is an enum class with enum values: `Supported` and `NotSupported`.

Reply:
This incurs 3! indirect loads. We shouldn't add code like that. This code is hot.